### ISSUE TITLE
Generate constructors

### ISSUE THREAD

**Issue by tjgurwara99** [golang/vscode-go#2507]
**Is your feature request related to a problem? Please describe.**

Feature Request. I was wondering if there are any plans to add the capability of generating constructors for a given type from the command palette. Similar to how the "generate interface stubs" (based on `impl` tool) or "generate unit tests" works.

**Describe the solution you'd like**

Often times, people write constructors such as `NewXxx` for type a `Xxx`. A similar utility to generate constructors exists in the GoLand IDE and I think it would help boost peoples productivity when working on Go codebases.

**Describe alternatives you've considered**

I currently have a working command line utility written in Go that does this exact same thing but as its a binary program I have to context switch to that and use the terminal to run the program (plus I whipped it up quickly to suit my needs so can't say its production quality). I believe it would benefit the Go community a lot if something like that exists in the official Go extension.

**Comment by hyangah** [comment]
Looks like the actual functionality is implemented by using another tool github.com/tjgurwara99/constr
I don't think it's a good idea to add another external tool dependency just for this feature. Instead, is it possible to add this feature in the gopls? (maybe as a code action or a custom command)

**Comment by tjgurwara99** [comment]
I wasn't saying that you should use my solution but rather it is a request to add something similar to what I already have in the official go extension because I sincerely believe it would be beneficial for a lot of people 😄

> actual functionality is implemented by using another tool github.com/tjgurwara99/constr

On this, I believe that my solution is no different from the tool `impl` that is being used to generate the interface stubs (look at the `Go: Generate interface stubs` command on the palette. If you believe that this kind of solution is not correct, then I would argue that `impl` also shouldn't be used either. Food for thought...

**Comment by hyangah** [comment]
Yes, `impl` is about to be removed from the dependency, in favor of https://github.com/golang/go/issues/37537

**Comment by tjgurwara99** [comment]
Ah that makes a lot of sense - thanks! Is there documentation on how to work with code actions in `gopls` anywhere? I couldn't find it anywhere - I did inspect the changes Marwan had implemented [here](https://go-review.googlesource.com/c/tools/+/274372) but I didn't understand a lot of it. So I'm guessing that it would be a non issue once I read the documentation 😄
